pkg/syncer/shared/namespace_test.go

```diff
@@ -0,0 +1,92 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shared
+
+import (
+	"testing"
+
+	"github.com/kcp-dev/logicalcluster"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPhysicalClusterNamespaceName(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		description         string
+		namespaceLocator    NamespaceLocator
+		workloadClusterName string
+		want                string
+	}
+	testCases := []testCase{
+		{
+			description: "downstream namespace for the default upstream namespace in logicalCluster root:myorg:myworkspace, workloadCluster cluster1",
+			namespaceLocator: NamespaceLocator{
+				LogicalCluster: logicalcluster.New("root:myorg:myworkspace"),
+				Namespace:      "default",
+			},
+			workloadClusterName: "cluster1",
+			want:                "kcp-34kf8vte1jm8",
+		},
+		{
+			description: "downstream namespace for the default upstream namespace in logicalCluster root:myorg:myworkspace, workloadCluster cluster2, expect no collision",
+			namespaceLocator: NamespaceLocator{
+				LogicalCluster: logicalcluster.New("root:myorg:myworkspace"),
+				Namespace:      "default",
+			},
+			workloadClusterName: "cluster2",
+			want:                "kcp-37yxpmg5tadl",
+		},
+		{
+			description: "downstream namespace for a namespaceLocator with a really long namespace name, expect a 14 char hash",
+			namespaceLocator: NamespaceLocator{
+				LogicalCluster: logicalcluster.New("root:myorg:myworkspace"),
+				Namespace:      "realllllllllylooooooooongnameeeeeeespace",
+			},
+			workloadClusterName: "cluster1",
+			want:                "kcp-2c2j4vsm39gt",
+		},
+		{
+			description: "downstream namespace for a namespaceLocator with a really long logicalCluster name, expect a 14 char hash",
+			namespaceLocator: NamespaceLocator{
+				LogicalCluster: logicalcluster.New("root:myorg:myworkspace:andthisreallylonglogicalcluster"),
+				Namespace:      "namespace",
+			},
+			workloadClusterName: "cluster1",
+			want:                "kcp-og59d6505q52",
+		},
+	}
+	var previousNamespaces []string
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			got, err := PhysicalClusterNamespaceName(tc.namespaceLocator, tc.workloadClusterName)
+			require.NoError(t, err)
+			require.Equal(t, tc.want, got)
+			// Make sure we don't produce namespace that are too long,
+			// as other components use the namespace as part of their
+			// generated resources (e.g. Ingresses, Routes).
+			// See issue: https://github.com/kcp-dev/kcp/issues/1188
+			if len(got) > 16 {
+				t.Errorf("namespace name is too long: %s", got)
+			}
+			// Make sure we have no collisions.
+			for _, ns := range previousNamespaces {
+				require.NotEqual(t, got, ns)
```

Comment:
not a really a helpful test, is it?

Reply:
not really.. I can drop that "previousNamespaces" check.